src/aws-fsx/ontap-file-system.ts

```diff
@@ -0,0 +1,498 @@
+    if (!/[a-zA-Z]/.test(fsxAdminPassword) || !/\d/.test(fsxAdminPassword) || /admin/i.test(fsxAdminPassword)) {
+      throw new Error(
+        "'fsxAdminPassword' must contain at least one English letter and one number, and must not contain the word 'admin'",
+      );
+    }
+  }
+
+  private validateHaPairs(deploymentType: OntapDeploymentType, haPairs?: number): void {
+    if (haPairs == null) {
+      return;
+    }
+    if (!Number.isInteger(haPairs) || haPairs < 1 || haPairs > 12) {
+      throw new Error(`\'haPairs\' must be an integer between 1 and 12, got ${haPairs}`);
+    }
+    if (
+      haPairs > 1 &&
+      [OntapDeploymentType.SINGLE_AZ_1, OntapDeploymentType.MULTI_AZ_1, OntapDeploymentType.MULTI_AZ_2].includes(
+        deploymentType,
+      )
+    ) {
+      throw new Error(`\'haPairs\' must be 1 for deployment type ${deploymentType}`);
+    }
+  }
+
+  private validateSubnets(
+    deploymentType: OntapDeploymentType,
+    vpcSubnets: aws_ec2.ISubnet[],
+    preferredSubnet?: aws_ec2.ISubnet,
+  ): void {
+    if (
+      (deploymentType === OntapDeploymentType.MULTI_AZ_1 || deploymentType === OntapDeploymentType.MULTI_AZ_2) &&
+      !preferredSubnet
+    ) {
+      throw new Error("'preferredSubnet' must be specified for multi-AZ file systems");
+    }
+    if (
+      (deploymentType === OntapDeploymentType.SINGLE_AZ_1 || deploymentType === OntapDeploymentType.SINGLE_AZ_2) &&
+      preferredSubnet
+    ) {
+      throw new Error("'preferredSubnet' can only be specified for multi-AZ file systems");
+    }
+    if (preferredSubnet && !vpcSubnets.includes(preferredSubnet)) {
+      throw new Error("'preferredSubnet' must be one of the specified 'vpcSubnets'");
+    }
+  }
+
+  private validateRouteTables(deploymentType: OntapDeploymentType, routeTables?: aws_ec2.IRouteTable[]): void {
+    if (routeTables == null || routeTables.length === 0) {
+      return;
+    }
+    if (deploymentType !== OntapDeploymentType.MULTI_AZ_1 && deploymentType !== OntapDeploymentType.MULTI_AZ_2) {
+      throw new Error("'routeTables' can only be specified for multi-AZ file systems");
+    }
+  }
+
+  private validateThroughputCapacity(
+    deploymentType: OntapDeploymentType,
+    throughputCapacity?: number,
+    throughputCapacityPerHaPair?: number,
+    haPair: number = 1,
+  ): void {
+    if (throughputCapacity == null && throughputCapacityPerHaPair == null) {
+      return;
+    }
+    if (throughputCapacity != null && throughputCapacityPerHaPair != null) {
+      throw new Error("'throughputCapacity' and 'throughputCapacityPerHaPair' cannot be specified at the same time");
+    }
+
+    // Calculate the throughput per HaPair and use it for validation,
+    // regardless of whether `throughputCapacity` or `throughputCapacityPerHaPair` is defined.
+    const throughputPerHaPair = throughputCapacityPerHaPair ?? throughputCapacity! / haPair;
```

Comment:
I want to pay special attention to the divisor.

Although the `validateHaPairs` method checks that `haPair` is greater than or equal to 1, this is a separate method and the situation could easily change. If other developers change the order of the validation methods in the future or make changes to the way the `haPair` is checked in the `validateHaPairs` method, can we guarantee that there will always be no assignment of 0? If not, the CDK will panic in this area.

In TypeScript, division by 0 is not an error, but when converted to other languages by jsii, there must be languages where it is an error (e.g. Golang).

Since this project specifically wants to guarantee safety and reliability, how about checking that the `haPair` is greater than 0 in this part as well? Although this case may not be necessary in most cases, it is better to be vigilant.

What do you think? Even if you think there is no need for this at all, at least adding a comment (that there is no possibility of 0 being passed on) to the code would be sufficient precaution. (There may be many developers like me who are sensitive to division by zero ...)

If you know of a better way, let me know.

Reply:
I'm completely agree with your opinion! I've added below code to validate zero division.

```ts
    if (haPair <= 0 || !Number.isInteger(haPair)) {
      throw new Error(`'haPair' must be a positive integer, got ${haPair}`);
    }
```